source/application/AMDSCentralServerSample.cpp

```diff
@@ -17,69 +17,72 @@ AMDSCentralServerSample::AMDSCentralServerSample(QObject *parent) :
 {
 }
 
+AMDSCentralServerSample::~AMDSCentralServerSample()
+{
+	if (fiftyMillisecondTimer_->isActive())
+		fiftyMillisecondTimer_->stop();
+	if (hundredMillisecondTimer_->isActive())
+		hundredMillisecondTimer_->stop();
+
+	fiftyMillisecondTimer_->deleteLater();
+	hundredMillisecondTimer_->deleteLater();
+}
 
 void AMDSCentralServerSample::onFiftyMillisecondTimerUpdate(){
 	AMDSLightWeightScalarDataHolder *oneScalerDataHolder = new AMDSLightWeightScalarDataHolder();
 	double valueAsDouble = (double)simpleCounter_++;
 	oneScalerDataHolder->setSingleValue(valueAsDouble);
 
-	energyBufferGroup_->append(oneScalerDataHolder);
+	bufferGroupManagers_.value("Energy")->append(oneScalerDataHolder);
 }
 
 void AMDSCentralServerSample::onHundredMillisecondTimerUpdate(){
-	AMDSLightWeightSpectralDataHolder *oneSpectralDataHolder = new AMDSLightWeightSpectralDataHolder(AMDSDataTypeDefinitions::Signed64, amptek1BufferGroup_->bufferGroupInfo().size(0));
+	AMDSLightWeightSpectralDataHolder *oneSpectralDataHolder = new AMDSLightWeightSpectralDataHolder(AMDSDataTypeDefinitions::Signed64, bufferGroupManagers_.value("Amptek1")->bufferGroupInfo().size(0));
 
-	AMDSFlatArray oneSpectralFlatArray(AMDSDataTypeDefinitions::Signed64, amptek1BufferGroup_->bufferGroupInfo().size(0));
+	AMDSFlatArray oneSpectralFlatArray(AMDSDataTypeDefinitions::Signed64, bufferGroupManagers_.value("Amptek1")->bufferGroupInfo().size(0));
 	spectralCounter_++;
 	qint64 oneValue;
 	for(int x = 0; x < oneSpectralFlatArray.constVectorQint64().size(); x++){
 		oneValue = (spectralCounter_+x)*2;
-		oneSpectralFlatArray.vectorQint64()[x] = oneValue;
+		oneSpectralFlatArray.setValue(x, oneValue);
 	}
 	oneSpectralDataHolder->setData(&oneSpectralFlatArray);
 
-	amptek1BufferGroup_->append(oneSpectralDataHolder);
+	bufferGroupManagers_.value("Amptek1")->append(oneSpectralDataHolder);
```

Comment:
Does something deal with the `bufferGroupManagers_` memory?  It appears like the light weight spectral holders are new'ed and added but not cleaned.